pkg/controllers/workgenerator/controller.go

```diff
@@ -0,0 +1,347 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+// Package workgenerator features a controller to generate work objects based on resource binding objects.
+package workgenerator
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/pointer"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+	workv1alpha1 "sigs.k8s.io/work-api/pkg/apis/v1alpha1"
+
+	fleetv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+	"go.goms.io/fleet/pkg/utils"
+	"go.goms.io/fleet/pkg/utils/controller"
+)
+
+// Reconciler watches binding objects and generate work objects in the designated cluster namespace
+// according to the information in the binding objects.
+// TODO: incorporate an overriding policy if one exists
+type Reconciler struct {
+	client.Client
+	recorder record.EventRecorder
+}
+
+// Reconcile triggers a single binding reconcile round.
+func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	klog.V(2).InfoS("Start to reconcile a ClusterResourceBinding", "cluster resource binding", req.Name)
+	startTime := time.Now()
+	bindingRef := klog.KRef(req.Namespace, req.Name)
+	// add latency log
+	defer func() {
+		klog.V(2).InfoS("ClusterResourceBinding reconciliation loop ends", "resource binding", bindingRef, "latency", time.Since(startTime).Milliseconds())
```

Comment:
```suggestion
		klog.V(2).InfoS("ClusterResourceBinding reconciliation loop ends", "resourceBinding", bindingRef, "latency", time.Since(startTime).Milliseconds())
```